packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -504,73 +504,120 @@ export class FungibleTokenTransferController implements ReactiveController {
+            !address ||
+            !provider ||
+            providerChainId !== this.sourceNetwork.chainId
+          ) {
+            this.estimatedGas = undefined;
+            this.resetFee();
+            return;
+          }
+          const {
+            pendingEvmApprovalTransactions,
+            pendingTransferTransaction,
+            fee,
+            resourceAmount
+          } = await this.buildEvmTransactions({
+            address,
+            chainId: this.destinationNetwork.chainId,
+            destinationAddress: this.destinationAddress,
+            resourceId: this.selectedResource.resourceId,
+            resourceAmount: this.resourceAmount,
+            provider,
+            providerChainId,
+            env: this.env,
+            fee: this.fee,
+            pendingEvmApprovalTransactions: this.pendingEvmApprovalTransactions,
+            pendingTransferTransaction: this
+              .pendingTransferTransaction as UnsignedTransaction,
+            sourceNetwork: this.sourceNetwork
+          });
+
+          this.fee = fee;
+          this.resourceAmount = resourceAmount;
+
+          this.pendingEvmApprovalTransactions = pendingEvmApprovalTransactions;
+          this.pendingTransferTransaction = pendingTransferTransaction;
+
+          await this.estimateGas(this.sourceNetwork);
+
+          this.host.requestUpdate();
         }
         break;
       case Network.SUBSTRATE:
         {
-          void this.buildSubstrateTransactions();
+          const substrateProvider = this.sourceSubstrateProvider;
+          const address =
+            this.walletContext.value?.substrateWallet?.signerAddress;
+
+          if (!substrateProvider || !address) {
+            this.estimatedGas = undefined;
+            this.resetFee();
+            return;
+          }
+          const { pendingTransferTransaction, fee, resourceAmount } =
+            await this.buildSubstrateTransactions({
+              address,
+              substrateProvider,
+              env: this.env,
+              chainId: this.destinationNetwork.chainId,
+              destinationAddress: this.destinationAddress,
+              resourceId: this.selectedResource.resourceId,
+              resourceAmount: this.resourceAmount,
+              fee: this.fee as SubstrateFee,
+              pendingTransferTransaction: this
+                .pendingTransferTransaction as SubstrateTransaction
+            });
+
+          this.fee = fee;
+          this.resourceAmount = resourceAmount;
+          this.pendingTransferTransaction = pendingTransferTransaction;
+
+          await this.estimateGas(this.sourceNetwork);
+          this.host.requestUpdate();
         }
         break;
       default:
         throw new Error('Unsupported network type');
     }
   }
 
-  public async estimateGas(): Promise<void> {
-    if (!this.sourceNetwork) return;
-    switch (this.sourceNetwork.type) {
+  public async estimateGas(sourceNetwork: Domain): Promise<void> {
```

Comment:
whats the reason of adding a `sourceNetwork` param? cause later in code you are basically checking `this.sourceNetwork?.chainId`

Reply:
removed